pkg/commands/patch/transform.go

```diff
@@ -61,6 +72,18 @@ func (self *patchTransformer) transformHeader() []string {
 			"--- a/" + self.opts.FileNameOverride,
 			"+++ b/" + self.opts.FileNameOverride,
 		}
+	} else if self.opts.TurnAddedFilesIntoDiffAgainstEmptyFile {
+		result := make([]string, 0, len(self.patch.header))
+		for idx, line := range self.patch.header {
+			if strings.HasPrefix(line, "new file mode") {
+				continue
+			}
+			if line == "--- /dev/null" && strings.HasPrefix(self.patch.header[idx+1], "+++ b/") {
+				line = "--- a/" + self.patch.header[idx+1][6:]
+			}
+			result = append(result, line)
```

Comment:
For the transformation to be really accurate, we'd also have to transform the line `index 0000000..fa1afe1` into `index e69de29..fa1afe1`, where "e69de29" is the hash of an empty blob. That's a bit tricky though, because that blob might not actually exist in the repo's object database (if the repo doesn't contain any empty files), so we'd have to call `git hash-object -w` to insert it, but we don't have access to git commands from here.

In practice, git seems to work just fine with the 00000 hash, even when conflicts are involved, as the test `move_to_index_from_added_file_with_conflict.go` shows, so I'm leaving it untouched.